server/blob/blob-cassandra/src/test/java/org/apache/james/blob/cassandra/cache/CachedBlobStoreTest.java

```diff
@@ -86,116 +88,190 @@ public BlobStore testee() {
 
         SoftAssertions.assertSoftly(soflty -> {
             assertThat(new ByteArrayInputStream(Mono.from(cache.read(blobId)).block()))
                 .hasSameContentAs(new ByteArrayInputStream(APPROXIMATELY_FIVE_KILOBYTES));
-            assertThat(new ByteArrayInputStream(Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block()))
+            assertThat(new ByteArrayInputStream(Mono.from(backend.readBytes(DEFAULT_BUCKETNAME, blobId)).block()))
                 .hasSameContentAs(new ByteArrayInputStream(APPROXIMATELY_FIVE_KILOBYTES));
         });
     }
 
     @Test
     public void shouldCacheWhenFiveKilobytesBytes() {
-        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, APPROXIMATELY_FIVE_KILOBYTES, SIZE_BASED)).block();
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKETNAME, APPROXIMATELY_FIVE_KILOBYTES, SIZE_BASED)).block();
 
         SoftAssertions.assertSoftly(soflty -> {
             assertThat(new ByteArrayInputStream(Mono.from(cache.read(blobId)).block()))
                 .hasSameContentAs(new ByteArrayInputStream(APPROXIMATELY_FIVE_KILOBYTES));
-            assertThat(new ByteArrayInputStream(Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block()))
+            assertThat(new ByteArrayInputStream(Mono.from(backend.readBytes(DEFAULT_BUCKETNAME, blobId)).block()))
                 .hasSameContentAs(new ByteArrayInputStream(APPROXIMATELY_FIVE_KILOBYTES));
         });
     }
 
     @Test
     public void shouldRemoveBothInCacheAndBackendWhenDefaultBucketName() {
-        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKERNAME, EIGHT_KILOBYTES, SIZE_BASED)).block();
+        BlobId blobId = Mono.from(testee().save(DEFAULT_BUCKETNAME, EIGHT_KILOBYTES, SIZE_BASED)).block();
 
         SoftAssertions.assertSoftly(ignored -> {
-            assertThatCode(Mono.from(testee().delete(DEFAULT_BUCKERNAME, blobId))::block)
+            assertThatCode(Mono.from(testee().delete(DEFAULT_BUCKETNAME, blobId))::block)
                 .doesNotThrowAnyException();
             assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
-            assertThatThrownBy(() -> Mono.from(backend.readBytes(DEFAULT_BUCKERNAME, blobId)).block())
+            assertThatThrownBy(() -> Mono.from(backend.readBytes(DEFAULT_BUCKETNAME, blobId)).block())
                 .isInstanceOf(ObjectNotFoundException.class);
         });
     }
+
+    @Test
+    public void shouldCacheWhenReadBytesWithDefaultBucket() {
+        BlobId blobId = Mono.from(backend.save(DEFAULT_BUCKETNAME, APPROXIMATELY_FIVE_KILOBYTES, SIZE_BASED)).block();
+
+        SoftAssertions.assertSoftly(soflty -> {
+            assertThat(Mono.from(cache.read(blobId)).blockOptional()).isEmpty();
+            assertThat(new ByteArrayInputStream(Mono.from(testee().readBytes(DEFAULT_BUCKETNAME, blobId)).block()))
```

Comment:
why do you need these conversion to ByteArrayInputStream?

Reply:
byte array comparison is expensive with assertj, imputStream comparisons:

 - is faster
 - Brings better error message

Reply:
I'm very surprise, do you know why it's more expensive? (I guess it tries to give more detailed messages but you say the opposite)

Reply:
I'm not fluent at reading base64 for each bytes of a byte array. I prefer something telling me "the content is different".

> I'm very surprise, do you know why it's more expensive? 

Hardly learnt experience. Anyway, the most efficient is hash comparison but it only makes sense at a few MB.

BTW I consider such concerns as not really relevant low level cosmetic concerns.

Reply:
> I'm not fluent at reading base64 for each bytes of a byte array. I prefer something telling me "the content is different".

I don't understand.

> Hardly learnt experience. Anyway, the most efficient is hash comparison but it only makes sense at a few MB.
> 
> BTW I consider such concerns as not really relevant low level cosmetic concerns.

If it's "Hardly learnt experience" how do you expect to share it if we don't discuss it here?

Reply:
> If it's "Hardly learnt experience" how do you expect to share it if we don't discuss it here?

That's what I'm currently doing?

> >  I'm not fluent at reading base64 for each bytes of a byte array. I prefer something telling me "the content is different".

> I don't understand.

Intellij prints out the whole byte arrays when prompting us the failure.

I find the "content of two stream is different" friendlier, and easier to work with

Reply:
Before:

![Capture d’écran de 2020-04-27 14-45-22](https://user-images.githubusercontent.com/6928740/80347270-4a39df00-8896-11ea-8b98-af73ceb73557.png)

After:

![Capture d’écran de 2020-04-27 14-48-28](https://user-images.githubusercontent.com/6928740/80347283-4efe9300-8896-11ea-8fe6-9b3b70361308.png)

Reply:
(the second version don't print me raw bytes hence I'm happier :-) )

Reply:
> > I'm not fluent at reading base64 for each bytes of a byte array. I prefer something telling me "the content is different".
> 
> I don't understand.
> 
> > Hardly learnt experience. Anyway, the most efficient is hash comparison but it only makes sense at a few MB.
> > BTW I consider such concerns as not really relevant low level cosmetic concerns.
> 
> If it's "Hardly learnt experience" how do you expect to share it if we don't discuss it here?

Yet you find that not relevant or I missed something

Reply:
I don't have issue explaining things.

I have an issue when providing explanation over cosmetic details delays overall work adoption.

What is not really the case here.

Sorry I must express terribly badly given that I need to explain the same thing ten time to be intelligible... At least it is the impression I have.